lib/node_modules/@stdlib/time/ms2duration/lib/main.js

```diff
@@ -0,0 +1,77 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isNonNegativeInteger = require( '@stdlib/assert/is-nonnegative-integer' ).isPrimitive;
+
+
+// MAIN //
+
+/**
+* Converts a number of milliseconds to a string duration.
+*
+* @param {number} ms - number of milliseconds
+* @throws {TypeError} must provide a nonnegative integer
+* @returns {string} duration
+*
+* @example
+* var str = ms2duration( 1030 );
+* // returns '1s30ms'
+*
+* @example
+* var str = ms2duration( 3600000 );
+* // returns '1h'
+*/
+function ms2duration( ms ) {
+	var out;
+	if ( !isNonNegativeInteger( ms ) ) {
+		throw new TypeError( 'invalid argument. Must provide a nonnegative integer. Value: `' + ms + '`.' );
+	}
+	if ( ms === 0 ) {
+		return '0ms';
+	}
+	out = '';
+	if ( ms >= 86400000 ) {
+		out += (ms/86400000|0) + 'd';
```

Comment:
We should use `floor` here. This only works is the result of division fits within a 32-bit int. Should also consider what to do if number serializes in exponential notation.

In short, values should always round-trip. Based on what I know of the inverse implementation, handling exponential notation is not supported (nor should it be).

Reply:
Changed to use `floor`.